aten/src/ATen/native/Blas.cpp

```diff
@@ -137,6 +137,18 @@ Tensor dot(const Tensor &self, const Tensor &other){
 
   dot_check(self, other);
 
+  if (self.is_complex()) {
+    if (self.is_conj()) {
+      if (other.is_conj()) {
+        return at::vdot(self.conj(), other.clone());
```

Comment:
since you're redispatching to calls with error checking, it would be better to do the rerouting *before* you do error checking

Reply:
Instead of cloning here, you should simply conjugate the result. dot(conj(A), conj(B)) = conj(dot(A, B)) (via distributivity of complex conjugation). As the output tensor is typically much smaller than the inputs this will be faster.

Reply:
makes sense! 
what do you think about returning `tensor.conj()` vs `tensor.conj_physical()`. I worry that it might be a little unexpected for the users to get a tensor with conj bit set to `True` or `False` depending on whether one or more input tensors had their conj bit set to `True`. But it'll be faster and we did say that we want to minimize materializing the conjugation as much as possible and benefit from the conj bit optimizations wherever we can.

Reply:
@Lezcano may have some thoughts here, but in this case conj physical seems pretty acceptable, tbh

Reply:
fwiw, given that the observable behaviour of tensors does not depend on the conjugate bit, I lean towards returning tensors with the conj bit on. The more operations we have that take advantage of this, the less copies we will make, which is quite nice.

Reply:
For `dot`/`vdot` tbg like @ezyang I think it doesn't matter much whether we are returning an output with conjugate bit set or not, it's a scalar tensor, but for other ops returning tensors with conjugate bit set seems right (in the bright future, tensors with conjugate bit set will be first class citizens and won't need to be materialized, it's similar to how we are *not* returning contiguous tensors always, and instead propagate input strides).

Reply:
One reason to advocate for returning the conjugate bit set is because dot/vdot could be batched, and then in that case there is a nontrivial perf difference. Well, maybe consistency is the important thing.

Reply:
But if `dot`/`vdot` are batched they'll go through gemm/bmm, and the output will have conj bit set as returned by those ops? For consistency probably we have to have the same behavior, that's right.